### ISSUE TITLE
Peer dependency warning

### ISSUE THREAD

**Issue by mromarreyes** [itinance/react-native-fs#439]
When installing the package I get the following error:

npm WARN react-native-fs@2.9.10 requires a peer of react-native@^0.51.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-native-fs@2.9.10 requires a peer of prop-types@15.6.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-native-fs@2.9.10 requires a peer of create-react-class@^15.6.2 but none is installed. You must install peer dependencies yourself.

I'm running:
"react": "16.2.0",
"react-native": "0.52.2",

How do I fix?

**Comment by sbordov** [comment]
I'm having the same issue. If I downgrade to react-native@^0.51.0, I then get the following error:
npm WARN react-native@0.51.0 requires a peer of react@16.0.0 but none was installed.

If I downgrade react from 16.2.0 to 16.0.0, I get this lovely error:
npm WARN react-native-fs@2.9.10 requires a peer of react@^16.2.0 but none was installed.

Any help with this issue would be greatly appreciated.

**Comment by itinance** [comment]
Don't worry! These are WARNINGS, no Errors! Similar warnings appear on "react-native init testApp". It has something to do with required versions in package.json and we need another approach here towards RN upgrades. But nothing will break here.

**Comment by no23reason** [comment]
@itinance I agree, they are only warnings, however the `create-react-class` `peerDependecy` is imo unnecessary, as this package is used only in tests – it should only be a `devDependecy` (it already is).

**Comment by itinance** [comment]
@no23reason please feel always free to send a PR :)

**Comment by no23reason** [comment]
@itinance PR opened (#458) :)

**Comment by aMarCruz** [comment]
@itinance , 0.51.x is very restrictive, remember the caret fixes the the left-most non-zero digits (the minor version here: 51). Better use something like this:

```
  "devDependencies": {
    "react-native": ">=0.51 <0.54"
  },
```

ref: https://docs.npmjs.com/misc/semver

**Comment by no23reason** [comment]
@aMarCruz why `0.60` as the upper bound? React Native is numbered sequentially, I think the correct spec would be something like
```
"devDependencies": {
    "react-native": ">=0.51 <1"
},
```
However, one of the reasons React Native is not yet 1.x is that there can (and often are) breaking changes between minor releases. Therefore I believe a strict `peerDependency` spec is a good thing here as it tells you "This has been tested on this release of React Native and it works. Any other might not work properly".

**Comment by aMarCruz** [comment]
@no23reason yes. I'm pretty sure that a library that works with RN 0.51 will hardly work with RN 0.60.

If RN followed the Semver that range would be adequate, but a new version of RN always implies a "crossing of the fingers" and hope that things do not break ... ...not a lot :smirk:

**Comment by JBuijgers** [comment]
Also getting peer dependency warnings, although not the same, creating a new issue for this seemed a bit too much.
```
warning " > react-native-fs@2.15.3" has incorrect peer dependency "react-native@^0.59.5".
warning " > react-native-fs@2.15.3" has unmet peer dependency "react-native-windows@^0.57.2".
```

package.json
```
"react": "16.8.6",
"react-native": "0.60.6"
```

**Comment by Rits1272** [comment]
@JBuijgers getting the same issue. Unable to install an app on my device

**Comment by vicary** [comment]
npm forces the caret selector `^` whichever package you touched with it.

I mean, it is frustrating if #481 is unintentionally reverted in 587a6f7d9a16b4c6cfaca5457e966c4bfe64d3bb.

cc @itinance
